esphome/components/adc/sensor.py

```diff
@@ -119,12 +120,23 @@ def validate_adc_pin(value):
     raise NotImplementedError
 
 
+def validate_config(config):
+    if (
+        CONF_RAW in config
+        and config[CONF_RAW] is True
+        and CONF_ATTENUATION in config
+        and config[CONF_ATTENUATION] == "auto"
+    ):
```

Comment:
It's not necessary to check if `CONF_RAW` is present as it has a default, and for readability:
```suggestion
    if config[CONF_RAW] and config.get(CONF_ATTENUATION, None) == "auto":
```